src/routes/+layout.svelte

```diff
@@ -2,14 +2,22 @@
 	import { auth } from '../lib/firebase/firebase';
 	import { onMount } from 'svelte';
 	import { goto } from '$app/navigation';
-
+	import { authStore } from '../lib/store/store';
+	import {authHandlers} from '../lib/handlers/auth'
 	const noAuthRoutes = ['/', '/login', '/signup'];
 	let currentPath;
 	onMount(() => {
 		const unsubscribe = auth.onAuthStateChanged(async (user) => {
 			currentPath = window.location.pathname;
 			if (!user && !noAuthRoutes.includes(currentPath)) {
 				// Unauthenticated user wants to go to a protected route
+				authStore.update((curr) => {
```

Comment:
Moving the auth state updates outside of if blocks seems like a better idea to me, what do you think?
Here, we are not updating the state in the second block. There is also repetition. It'd be better to extract a function that takes user as parameter, and call it on each auth state change.

Reply:
Thank you for the suggestion! I created a function and now call it on auth state change

Reply:
Looks great! We might want to move the function to another file like helpers, but for now this will do great!